scripts/sct_analyze_lesion.py

```diff
@@ -104,410 +101,123 @@ def get_parser():
-            # Compute mean and std value in each labeled lesion
-            self._measure_within_im(im_lesion=im_lesion_data, im_ref=Image(self.fname_ref).data, label_lst=label_lst)
-
-    def _normalize(self, vect):
-        norm = np.linalg.norm(vect)
-        return vect / norm
-
-    def angle_correction(self):
-        im_seg = Image(self.fname_sc)
-        nx, ny, nz, nt, px, py, pz, pt = im_seg.dim
-        data_seg = im_seg.data
-
-        # fit centerline, smooth it and return the first derivative (in physical space)
-        _, arr_ctl, arr_ctl_der, _ = get_centerline(im_seg, param=ParamCenterline(), verbose=1)
-        x_centerline_deriv, y_centerline_deriv, z_centerline_deriv = arr_ctl_der
-
-        self.angles = np.full_like(np.empty(nz), np.nan, dtype=np.double)
-
-        # loop across x_centerline_deriv (instead of [min_z_index, max_z_index], which could vary after interpolation)
-        for iz in range(x_centerline_deriv.shape[0]):
-            # normalize the tangent vector to the centerline (i.e. its derivative)
-            tangent_vect = self._normalize(np.array(
-                [x_centerline_deriv[iz] * px, y_centerline_deriv[iz] * py, pz]))
-
-            # compute the angle between the normal vector of the plane and the vector z
-            angle = np.arccos(np.vdot(tangent_vect, np.array([0, 0, 1])))
-            self.angles[iz] = math.degrees(angle)
-
-    def label_lesion(self):
-        printv('\nLabel connected regions of the masked image...', self.verbose, 'normal')
-        im = Image(self.fname_mask)
-        im_2save = im.copy()
-        im_2save.data = label(im.data, connectivity=2)
-        im_2save.save(self.fname_label)
-
-        self.measure_pd['label'] = [l for l in np.unique(im_2save.data) if l]
-        printv('Lesion count = ' + str(len(self.measure_pd['label'])), self.verbose, 'info')
-
-    def _orient(self, fname, orientation):
-        return Image(fname).change_orientation(orientation).save(fname, mutable=True)
-
-    def orient2rpi(self):
-        # save input image orientation
-        self.orientation = Image(self.fname_mask).orientation
-
-        if not self.orientation == 'RPI':
-            printv('\nOrient input image(s) to RPI orientation...', self.verbose, 'normal')
-            self._orient(self.fname_mask, 'RPI')
-
-            if self.fname_sc is not None:
-                self._orient(self.fname_sc, 'RPI')
-            if self.fname_ref is not None:
-                self._orient(self.fname_ref, 'RPI')
-            if self.path_template is not None:
-                self._orient(self.path_levels, 'RPI')
-                for fname_atlas in self.atlas_roi_lst:
-                    self._orient(fname_atlas, 'RPI')
-
-    def ifolder2tmp(self):
-        # copy input image
-        if self.fname_mask is not None:
-            sct.copy(self.fname_mask, self.tmp_dir)
-            self.fname_mask = ''.join(extract_fname(self.fname_mask)[1:])
+def check_binary(arr, fname, verbose=1):
```

Comment:
this one could potentially go under `spinalcordtoolbox.utils` or `spinalcordtoolbox.image` (or else?)
@kousu any suggestion?